span-processing-config-service-impl/src/main/java/org/hypertrace/span/processing/config/service/SpanProcessingConfigServiceImpl.java

```diff
@@ -156,6 +209,77 @@ public void deleteExcludeSpanRule(
     }
   }
 
+  @SneakyThrows
+  private ExcludeSpanRule buildExcludeSpanRuleFromConfig(
+      com.typesafe.config.ConfigObject configObject) {
+    String jsonString = configObject.render();
+    ExcludeSpanRule.Builder builder = ExcludeSpanRule.newBuilder();
+    JsonFormat.parser().merge(jsonString, builder);
+    return builder.build();
+  }
+
+  private Optional<ExcludeSpanRule> getSystemExcludeSpanRule(String id) {
+    if (systemExcludeSpanRuleIdToRuleMap.containsKey(id)) {
+      return Optional.of(systemExcludeSpanRuleIdToRuleMap.get(id));
+    }
+    return Optional.empty();
+  }
+
+  private List<ExcludeSpanRuleDetails> reorderExcludeSpanRules(
+      List<ExcludeSpanRuleDetails> userExcludeSpanRules,
```

Comment:
If we convert list into a map, then we can directly use `Maps.difference(userExcludeSpanRuleIdToRuleMap, systemExcludeSpanRuleIdToRuleMap)`. You can use

`entriesInCommon` for overriden rules
`entriesOnLeft` for only user rules
`entriesOnRight` for only system rules

Reply:
But we would not be able to retain order among rules within the same groups right?🤔 Am I missing something?